models/exposure.py

```diff
@@ -1,7 +1,238 @@
 
-from models.base import Base
+import inspect
+from collections import defaultdict
 
+import numpy as np
+
+import sqlalchemy as sa
+from sqlalchemy import orm
+from sqlalchemy.dialects.postgresql import JSONB
+
+from astropy.coordinates import SkyCoord
+
+from models.base import Base, SpatiallyIndexed
+import models.instruments
+
+from pipeline.utils import normalize_header_key
+
+class CCD_Data:
+    """
+    A helper class that lazy loads the CCD data from the database.
+    When requesting one of the CCD IDs it will fetch that data from
+    disk and store it in memory.
+    To clear the memory cache, call the clear_cache() method.
+    """
+    def __init__(self, filename, inst):
+        self.filename = filename
+        self.instrument_object = inst
+        self._data = defaultdict(lambda: None)
+
+    def __getitem__(self, ccd_id):
+        if self._data[ccd_id] is None:
+            self._data[ccd_id] = self.instrument_object.load(self.filename, ccd_id)
+        return self._data[ccd_id]
+
+    def __setitem__(self, ccd_id, value):
+        self._data[ccd_id] = value
+
+    def clear_cache(self):
+        self._data = defaultdict(lambda: None)
+
+
+class Exposure(Base, SpatiallyIndexed):
 
-class Exposure(Base):
     __tablename__ = "exposures"
 
+    header = sa.Column(JSONB, nullable=False, default={}, doc="Header of the raw exposure. ")
+
+    filename = sa.Column(sa.Text, nullable=False, index=True, unique=True, doc="Filename for raw exposure. ")
+
+    mjd = sa.Column(
```

Comment:
Something we either have to think about, or have to decide not to think about, is the time standard we're going to use....   That is, TAI vs. UTC vs. etc. See, for instance : https://www.ipses.com/eng/in-depth-analysis/standard-of-time-definition/

They're all different by less than a minute, so my inclination is to decide not to think about this.  Just say "hey, whatever the header says the MJD is, that we're gonna believe", and as long as we're only looking at headers from the same telescope, it will all be consistent.